roxie/ccd/ccdcontext.cpp

```diff
@@ -3497,11 +3497,11 @@ class CRoxieServerContext : public CRoxieContextBase, implements IRoxieServerCon
     {
         CriticalBlock b(contextCrit);
         StringBuffer expandedName;
-        expandLogicalFilename(expandedName, fileName, workUnit, false, !workUnit);
+        expandLogicalFilename(expandedName, fileName, workUnit, false, !serverQueryFactory->isDynamic());
```

Comment:
If we get here we are definitely doing a dynamic filename lookup (as opposed to a pre-resolve). Whether we think that implies that we should always follow the foreign, I'm not sure. People CAN specify ,DYNAMIC on a filename in a deployed query - should that imply foreign is followed, or not?